File: tests/02_task_depend/test02_task_depend.py
#!/usr/bin/env python

import iris

iris.init(True)

task4 = iris.task_create("A")
iris.task_submit(task4, iris.iris_cpu, False);

task5 = iris.task_create("B")
iris.task_submit(task5, iris.iris_gpu, False);

task6 = iris.task_create("C")
task6_dep = [ task5 ]
iris.task_depend(task6, 1, task6_dep);
iris.task_submit(task6, iris.iris_cpu, False);

task7 = iris.task_create("D")
task7_dep = [ task4, task6 ]
iris.task_depend(task7, 2, task7_dep);
iris.task_submit(task7, iris.iris_gpu, False);

task8 = iris.task_create("E")
task8_dep = [ task5 ]
iris.task_depend(task8, 1, task8_dep);
iris.task_submit(task8, iris.iris_cpu, False);

iris.finalize()

File: tests/02_task_depend/test02_task_depend.c
#include <iris/iris.h>
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char** argv) {
  iris_init(&argc, &argv, true);

  iris_task task2;
  iris_task_create_name("task2", &task2);
  iris_task_submit(task2, iris_cpu, NULL, false);

  iris_task task3;
  iris_task_create_name("task3", &task3);
  iris_task_submit(task3, iris_gpu, NULL, false);

  iris_task task4;
  iris_task task4_dep[] = { task3 };
  iris_task_create_name("task4", &task4);
  iris_task_depend(task4, 1, task4_dep);
  iris_task_submit(task4, iris_cpu, NULL, false);

  iris_task task5;
  iris_task task5_dep[] = { task2, task4 };
  iris_task_create_name("task5", &task5);
  iris_task_depend(task5, 2, task5_dep);
  iris_task_submit(task5, iris_gpu, NULL, false);

  iris_task task6;
  iris_task task6_dep[] = { task2 };
  iris_task_create_name("task6", &task6);
  iris_task_depend(task6, 1, task6_dep);
  iris_task_submit(task6, iris_cpu, NULL, false);

  iris_finalize();

  int errors = iris_error_count();
  printf("Errors:%d\n", errors);
  return errors;
}
